apps/web/src/scenes/NftDetailPage/NftDetailLightbox.tsx

```diff
@@ -0,0 +1,141 @@
+import { useCallback, useEffect, useRef, useState } from 'react';
+import colors from 'shared/theme/colors';
+import styled, { css, keyframes } from 'styled-components';
+
+import { StyledContainer } from '~/contexts/modal/AnimatedModal';
+import CloseIcon from '~/icons/CloseIcon';
+
+type Props = {
+  toggleLightbox: () => void;
+  isLightboxOpen: boolean;
+  tokenId: string;
+};
+
+// This component is used to display an NFT in fullscreen.
+// It is responsible for the transition animations including sizing and fullscreen background.
+// To display the NFT itself, this component simply creates a portal destination in which another component (such as TokenDetailAsset) can place an NFT asset view inside.
+// This is necessary to use the exact same rendered asset when viewing it in on both fullscreen and non-fullscreen modes to achieve a seamless transition.
+export default function NftDetailLightbox({ toggleLightbox, isLightboxOpen, tokenId }: Props) {
+  const ref = useRef<HTMLDivElement>(null);
+  const [distanceFromLeft, setDistanceFromLeft] = useState(0);
+
+  // Because the lightbox is originally positioned absolutely relative to the displayed nft on the detail page, when the lightbox is opened to be fullscreen
+  // we have to manually position it to aligned with the edge of the browser window.
+  useEffect(() => {
+    if (ref.current) {
+      const rect = ref.current.getBoundingClientRect();
+      setDistanceFromLeft(rect.left);
+    }
+  }, []); // Empty dependency array means this runs once on mount
+
+  const handleCloseClick = useCallback(() => {
+    setShouldFadeOut(true);
+    toggleLightbox();
+  }, [toggleLightbox]);
+
+  // we have a separate state for the background because we need to delay unmounting it to show the fade out animation
+  const [showBackground, setShowBackground] = useState(isLightboxOpen);
+  const [shouldFadeOut, setShouldFadeOut] = useState(false);
+
+  useEffect(() => {
+    if (isLightboxOpen) {
+      setShowBackground(true);
+    } else {
+      setTimeout(() => {
+        setShowBackground(false);
+        setShouldFadeOut(false);
+      }, 500);
```

Comment:
is this `500` supposed to line up with the 0.5s listed in the CSS below? if so we should consolidate into a const

CSS allows you to use `ms` as well, so `0.5s` can be written as `500ms`

Reply:
giga brain